hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java

```diff
@@ -1558,8 +1559,8 @@ public List<OzoneKey> listKeys(String volumeName, String bucketName,
                                  String keyPrefix, String prevKey,
                                  int maxListResult)
       throws IOException {
-    List<OmKeyInfo> keys = ozoneManagerClient.listKeys(
-        volumeName, bucketName, prevKey, keyPrefix, maxListResult);
+    List<BasicOmKeyInfo> keys = ozoneManagerClient.listKeysLight(
```

Comment:
We will need to take into account the older OM servers which do not support the newer API similar to ```  private OmKeyInfo getKeyInfo(OmKeyArgs keyArgs) throws IOException {
    final OmKeyInfo keyInfo;
    if (omVersion.compareTo(OzoneManagerVersion.OPTIMIZED_GET_KEY_INFO) >= 0) {
      keyInfo = ozoneManagerClient.getKeyInfo(keyArgs, false)
          .getKeyInfo();
    } else {
      keyInfo = ozoneManagerClient.lookupKey(keyArgs);
    }
    return keyInfo;
  }```

Reply:
Thank you @kerneltime, I have addressed the write compatibility issue to make sure an older server version can work together with a newer client version.